internal/acceptance/model_serving_test.go

```diff
@@ -0,0 +1,116 @@
+package acceptance
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccModelServing(t *testing.T) {
+	name := fmt.Sprintf("terraform-test-model-serving-%[1]s",
+		acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum))
+	workspaceLevel(t, step{
+		Template: fmt.Sprintf(`
+		data "databricks_spark_version" "latest" {
+		}
+		resource "databricks_cluster" "this" {
+			cluster_name = "singlenode-{var.RANDOM}"
+			spark_version = data.databricks_spark_version.latest.id
+			instance_pool_id = "{env.TEST_INSTANCE_POOL_ID}"
+			num_workers = 0
+			autotermination_minutes = 10
+			spark_conf = {
+				"spark.databricks.cluster.profile" = "singleNode"
+				"spark.master" = "local[*]"
+			}
+			custom_tags = {
+				"ResourceClass" = "SingleNode"
+			}
+			library {
+				pypi {
+					package = "mlflow"
+				}
+			}
+		}
+		resource "databricks_mlflow_experiment" "exp" {
+			name = "/Shared/%[1]s-exp"
+		}
+		resource "databricks_mlflow_model" "model" {
+			name = "%[1]s-model"
+		}
+		`, name),
+		Check: func(s *terraform.State) error {
+			w := databricks.Must(databricks.NewWorkspaceClient())
+			id := s.RootModule().Resources["databricks_cluster.this"].Primary.ID
+			w.CommandExecutor.Execute(context.Background(), id, "python", fmt.Sprintf(`
+				import time
+				import mlflow
+				import mlflow.pyfunc
+				from mlflow.tracking.artifact_utils import get_artifact_uri
+				from mlflow.tracking.client import MlflowClient
+
+				mlflow.set_experiment("/Shared/%[1]s-exp")
+
+				class SampleModel(mlflow.pyfunc.PythonModel):
+					def predict(self, ctx, input_df):
+						return 7
+				artifact_path = 'sample_model'
+				
+				with mlflow.start_run() as new_run:
+					mlflow.pyfunc.log_model(python_model=SampleModel(), artifact_path=artifact_path)
+					run1_id = new_run.info.run_id
+					source = get_artifact_uri(run_id=run1_id, artifact_path=artifact_path)
+
+				client = MlflowClient()
+				client.create_model_version(name="%[1]s-model", source=source, run_id=run1_id)
+				client.create_model_version(name="%[1]s-model", source=source, run_id=run1_id)
+				time.sleep(100) # wait for model version to finish creation
```

Comment:
Nit: it's a little bit brittle, what other options do we have? A retry loop that checks if the model version is available? Would that be faster?

Reply:
I imagine the terraform endpoint resource wouldn't block until the model version is avalable, right?

Reply:
No it fails if the model version isn't available. The timeout that we have for model creation is 300 seconds, at which point the creation fails, so I could bump it to that but generally if it's going to succeed, it will succeed long before that.

Reply:
Yeah, I was thinking we could make this test both more reliable and faster with explicit retries. Sometimes it could exit the wait within 10 seconds, for instance, doesn't have to wait 100s everytime. Non blocker for this PR.

Reply:
i agree with @vladimirk-db comment, please rewrite to retries with smaller intervals, like every 10 seconds.

Reply:
Added retry logic